core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1924,6 +1924,22 @@ public static int truncate(int v, int x) {
     return v - remainder;
   }
 
+  /** SQL {@code LAST_DAY} function. */
+  public static int lastDay(int date) {
+    int y0 = (int) DateTimeUtils.unixDateExtract(TimeUnitRange.YEAR, date);
+    int m0 = (int) DateTimeUtils.unixDateExtract(TimeUnitRange.MONTH, date);
+    int last = lastDay(y0, m0);
+    return DateTimeUtils.ymdToUnixDate(y0, m0, last);
+  }
+
+  public static int lastDay(long timestamp) {
```

Comment:
Pls add javadoc for `timestamp` argument. Is it millis since epoch ?

Reply:
Ok.